generate-task-list.js

```diff
@@ -0,0 +1,55 @@
+const addTaskForm = document.querySelector('.task-form');
+const taskList = document.querySelector('.task-list');
+const addTask = event => {
+	event.preventDefault();
+
+	const taskName = document.querySelector('.task-form__input--task-content').value;
+	const taskDeadline = document.querySelector('.task-form__input--date').value;
+
+	let lastElement = null;
+	let lastElementIndex = null;
+
+	console.log(tasks.length);
+	if (tasks.length > 0) {
+		lastElementIndex = tasks.length - 1;
+		lastElement = tasks[lastElementIndex].name;
+	} else {
+		lastElement = tasks[0];
+		lastElementIndex = 0;
+	}
+	tasks.push({
+		id: lastElement.id + 1,
+		name: taskName,
+		deadline: taskDeadline,
+		isDone: false,
+		isDeleted: false
```

Comment:
I've thought it over and keeping separate boolean fields for `isDone` `isDeleted` is IMHO not the cleanest solution. It would be better to keep a single `status` field. Given task item can belong to exactly one list at the time, so it can be either `NEW`/`DONE` or `DELETED`.